service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/pis/PaymentCancellationSpiImpl.java

```diff
@@ -1,62 +1,110 @@
   public @NotNull SpiResponse<SpiPaymentCancellationResponse> initiatePaymentCancellation(
-      @NotNull SpiPsuData spiPsuData, @NotNull SpiPayment spiPayment,
+      @NotNull SpiPsuData spiPsuData,
+      @NotNull SpiPayment spiPayment,
       @NotNull AspspConsentData aspspConsentData) {
-    return null;
+    SpiPaymentCancellationResponse cancellationResponse = new SpiPaymentCancellationResponse();
+    cancellationResponse.setCancellationAuthorisationMandated(true);
+    cancellationResponse.setTransactionStatus(SpiTransactionStatus.valueOf(
+        paymentDataRepository.findByPaymentId(spiPayment.getPaymentId()).get().get(0)
+            .getTransactionStatus().name()));
+    return new SpiResponse<>(cancellationResponse, aspspConsentData);
   }
 
   @Override
   public @NotNull SpiResponse<SpiResponse.VoidResponse> cancelPaymentWithoutSca(
-      @NotNull SpiPsuData spiPsuData, @NotNull SpiPayment spiPayment,
+      @NotNull SpiPsuData spiPsuData,
+      @NotNull SpiPayment spiPayment,
       @NotNull AspspConsentData aspspConsentData) {
-    return null;
+    return new SpiResponse<>(SpiResponse.voidResponse(), aspspConsentData);
   }
 
   @Override
   public @NotNull SpiResponse<SpiResponse.VoidResponse> verifyScaAuthorisationAndCancelPayment(
-      @NotNull SpiPsuData spiPsuData, @NotNull SpiScaConfirmation spiScaConfirmation,
-      @NotNull SpiPayment spiPayment, @NotNull AspspConsentData aspspConsentData) {
-    return null;
+      @NotNull SpiPsuData spiPsuData,
+      @NotNull SpiScaConfirmation spiScaConfirmation,
+      @NotNull SpiPayment spiPayment,
+      @NotNull AspspConsentData aspspConsentData) {
+    if (spiScaConfirmation.getTanNumber().equals(MOCKED_TAN)) {
+      Optional<List<PisPaymentData>> paymentDataList = paymentDataRepository
+          .findByPaymentId(spiPayment.getPaymentId());
+
+      if (paymentDataList.isPresent()) {
+        PisPaymentData payment = paymentDataList.get().get(0);
+        payment.setTransactionStatus(TransactionStatus.CANC);
+        paymentDataRepository.save(payment);
+        return new SpiResponse<>(SpiResponse.voidResponse(), aspspConsentData);
+      }
+
+      return SpiResponse.<SpiResponse.VoidResponse>builder()
+          .aspspConsentData(aspspConsentData)
+          .message(Collections.singletonList("Payment not found"))
+          .fail(SpiResponseStatus.LOGICAL_FAILURE);
+    }
+
+    return SpiResponse.<SpiResponse.VoidResponse>builder()
+        .aspspConsentData(aspspConsentData)
+        .message(Collections.singletonList("Wrong PIN"))
+        .fail(SpiResponseStatus.UNAUTHORIZED_FAILURE);
   }
 
   @Override
   public SpiResponse<SpiAuthorisationStatus> authorisePsu(@NotNull SpiPsuData spiPsuData,
-      String s, SpiPayment spiPayment, AspspConsentData aspspConsentData) {
-    return null;
+      String password, SpiPayment spiPayment, AspspConsentData aspspConsentData) {
+    return this.authorisationService.authorisePsu(spiPsuData, password, aspspConsentData);
```

Comment:
There will be merge conflicts when you rebase your code, because the authorisationService gets passed an additional attribute `String iban` which is needed for the correct PSU-Validation.
You can probably reuse the code from Consent-Creation

Reply:
you are going to face the merge conflicts ;)